src/Analyser/NodeScopeResolver.php

```diff
@@ -6134,9 +6146,12 @@ private function isImplicitArrayCreation(array $dimFetchStack, Scope $scope): Tr
 	/**
 	 * @param list<ArrayDimFetch> $dimFetchStack
 	 * @param list<Type|null> $offsetTypes
+	 * @param list<array{Expr, Type}> $additionalExpressions
```

Comment:
I'm not a fan of by-ref parameters. I'd rather if the method returned `array{Type, list<array{Expr, Type}>}` 😊